src/Operation/EstimatedDocumentCount.php

```diff
@@ -90,18 +115,53 @@ public function __construct($databaseName, $collectionName, array $options = [])
      */
     public function execute(Server $server)
     {
-        return $this->count->execute($server);
+        $command = $this->createCommand($server);
+        if ($command instanceof Aggregate) {
+            try {
+                $cursor = $command->execute($server);
+            } catch (CommandException $e) {
+                if ($e->getCode() == self::$errorCodeCollectionNotFound) {
+                    return 0;
+                }
+
+                throw $e;
+            }
+
+            $cursor->rewind();
+
+            return $cursor->current()->n;
```

Comment:
Is this worth an `(integer)` cast just to be sure? I wouldn't expect anything else to be returned, but it is technically a wild value from the server.